core/src/main/java/com/linecorp/armeria/client/endpoint/healthcheck/HealthCheckedEndpointGroup.java

```diff
@@ -139,7 +139,8 @@ public static HealthCheckedEndpointGroupBuilder builder(EndpointGroup delegate,
 
         clientOptions.factory().whenClosed().thenRun(this::closeAsync);
         delegate.addListener(this::updateCandidates);
-        updateCandidates(delegate.whenReady().join());
+        delegate.whenReady().join();
+        updateCandidates(delegate.endpoints());
```

Comment:
Seems it is better to leave a comment the reason why we cannot use the result of `delegate.whenReady().join()`?

Reply:
Added. 😄